common.c

```diff
@@ -101,6 +123,9 @@ int start_listen_sockets(int *sockfd[], struct addrinfo *addr_list)
    }
 
    return num_addr;
+#ifdef SYSTEMD
+   }
+#endif
 }
```

Comment:
Ok, this is the only issue I have with the patch: mixing compilation directives within the code. It's ugly and makes the code really hard to read!
Please move the conditional code (lines 74-89) to another function which entire content will be conditional, and make start_listen_sockets call that unconditionally. There are lots of examples of that in sslh-main.c with the LIBCAP and LIBCONFIG symbols.